pkg/graveler/ref/manager.go

```diff
@@ -175,6 +175,20 @@ func (m *Manager) GetBranch(ctx context.Context, repositoryID graveler.Repositor
 	return branch.(*graveler.Branch), nil
 }
 
+func (m *Manager) CreateBranch(ctx context.Context, repositoryID graveler.RepositoryID, branchID graveler.BranchID, branch graveler.Branch) error {
+	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
+		_, err := tx.Exec(`
+			INSERT INTO graveler_branches (repository_id, id, staging_token, commit_id)
+			VALUES ($1, $2, $3, $4)`,
+			repositoryID, branchID, branch.StagingToken, branch.CommitID)
+		return nil, err
+	})
+	if errors.Is(err, db.ErrAlreadyExists) {
+		return graveler.ErrBranchExists
+	}
```

Comment:
IIUC this replaces `SetBranch`, which would "re-seat" the branch at the new location.  This is a cool fix, but I don't understand how this fixes any other calls on graveler racing with anything else that calls `SetBranch`, e.g. `Commit`.  Those are considerably more dangerous I reckon.

So I think the real issue remains #2405.

Reply:
True, this one will solve the creation of two branches creation in parallel - but not the locking issue based on the second connection pool.